integration/combination/test_function_with_deployment_preference.py

```diff
@@ -97,7 +102,9 @@ def _get_deployment_groups(self, application_name):
         ]
 
     def _get_deployments(self, application_name, deployment_group):
-        deployments = self.client_provider.code_deploy_client.list_deployments()["deployments"]
+        deployments = self.client_provider.code_deploy_client.list_deployments(
+            applicationName=application_name, deploymentGroupName=deployment_group
+        )["deployments"]
```

Comment:
For client responses like this are we catching any possible client errors? Also if the client returned a None response due to some error ["deployments"] would also be error prone? (Maybe get("deployments") would be a better option

Reply:
If it returns a None, it will break and the test will failed as expected(it shouldn't be none).